org.osgi.test.junit5.cm/src/main/java/org/osgi/test/junit5/cm/ConfigurationExtension.java

```diff
@@ -0,0 +1,407 @@
+
+
+		Configuration ic = ConfigUtil.getConfigsByServicePid(getConfigAdmin(), injectConfiguration.value(),
+			injectConfiguration.timeout());
+
+		Object objectToInject = transformConfigToTargetType(ic, targetType, getConfigAdmin());
+
+		setField(field, extensionContext.getRequiredTestInstance(), objectToInject);
+	}
+
+	private void storeConfigCopy(ExtensionContext extensionContext) throws Exception {
+
+		List<Configuration> configurations = ConfigUtil.getAllConfigurations(getConfigAdmin());
+		List<ConfigurationCopy> configurationCopies = ConfigUtil.cloneConfigurations(configurations);
+
+		extensionContext.getStore(Namespace.create(ConfigurationExtension.class, extensionContext.getUniqueId()))
+			.put(STORE_CONFIGURATION_KEY, configurationCopies);
+
+	}
+
+	Configuration handleWithConfiguration(WithConfiguration configAnnotation, ConfigurationAdmin configurationAdmin,
+		BlockingConfigurationHandler blockingConfigHandler)
+		throws ParameterResolutionException, IllegalArgumentException {
+
+		try {
+			Configuration configBefore = ConfigUtil.getConfigsByServicePid(configurationAdmin, configAnnotation.pid(),
+				0l);
+
+			Configuration configuration = configurationAdmin.getConfiguration(configAnnotation.pid());
+
+			updateConfigurationRespectNew(configuration, ConfigUtil.of(configAnnotation.properties()),
+				configBefore == null);
+
+			return configuration;
+
+		} catch (Exception e) {
+			throw new ParameterResolutionException("ConfigurationAdmin could not be found", e);
+		}
+
+	}
+
+	Configuration handleWithFactoryConfiguration(WithFactoryConfiguration configAnnotation)
+		throws ParameterResolutionException, IllegalArgumentException {
+
+		try {
+			ConfigurationAdmin configurationAdmin = getConfigAdmin();
+
+			Configuration configBefore = ConfigUtil.getConfigsByServicePid(configurationAdmin,
+				configAnnotation.factoryPid() + "~" + configAnnotation.name());
+
+			Configuration configuration = configurationAdmin.getFactoryConfiguration(configAnnotation.factoryPid(),
+				configAnnotation.name());
+
+			updateConfigurationRespectNew(configuration, ConfigUtil.of(configAnnotation.properties()),
+				configBefore == null);
+
+			return configuration;
+
+		} catch (Exception e) {
+			throw new ParameterResolutionException("ConfigurationAdmin could not be found", e);
+		}
+
+	}
+
+	static void assertValidFieldCandidate(Field field) {
+		if (Modifier.isFinal(field.getModifiers()) || Modifier.isPrivate(field.getModifiers())
+			|| Modifier.isStatic(field.getModifiers())) {
+			throw new ExtensionConfigurationException("@" + WithConfiguration.class.getSimpleName() + " field ["
+				+ field.getName() + "] must not be final, private or static.");
+		}
+	}
+
+	@Override
+	public Object resolveParameter(ParameterContext parameterContext, ExtensionContext extensionContext)
+		throws ParameterResolutionException {
+
+		try {
+
+			BundleContext bundleContext = FrameworkUtil.getBundle(extensionContext.getRequiredTestClass())
```

Comment:
You should not get the BundleContext directly. Use BundleContextExtension to get a BundleContext if needed.